packages/browser-wallet/src/shared/utils/payload-helpers.ts

```diff
@@ -1,34 +1,34 @@
 import {
+    AccountTransactionInput,
     AccountTransactionType,
-    SimpleTransferPayload,
-    UpdateContractPayload,
-    serializeUpdateContractParameters,
-    AccountTransactionPayload,
-    SchemaVersion,
-    InitContractPayload,
-    serializeInitContractParameters,
-    serializeTypeValue,
     ContractName,
     EntrypointName,
-    Parameter,
-    jsonParse,
     getAccountTransactionHandler,
+    InitContractInput,
+    jsonParse,
+    Parameter,
+    SchemaVersion,
+    serializeInitContractParameters,
+    serializeTypeValue,
+    serializeUpdateContractParameters,
+    SimpleTransferPayload,
+    UpdateContractInput,
 } from '@concordium/web-sdk';
 import { Buffer } from 'buffer/';
-import { SmartContractParameters, SchemaType, SchemaWithContext } from '@concordium/browser-wallet-api-helpers';
+import { SchemaType, SchemaWithContext, SmartContractParameters } from '@concordium/browser-wallet-api-helpers';
 import { serializationTypes } from '@wallet-api/constants';
 import * as JSONBig from 'json-bigint';
 
 export type HeadlessTransaction =
-    | { type: AccountTransactionType.Update; payload: UpdateContractPayload }
+    | { type: AccountTransactionType.Update; payload: UpdateContractInput }
     | { type: AccountTransactionType.Transfer; payload: SimpleTransferPayload }
-    | { type: AccountTransactionType.InitContract; payload: InitContractPayload }
+    | { type: AccountTransactionType.InitContract; payload: InitContractInput }
     | {
           type: Exclude<
               AccountTransactionType,
               AccountTransactionType.Transfer | AccountTransactionType.Update | AccountTransactionType.InitContract
           >;
-          payload: AccountTransactionPayload;
+          payload: AccountTransactionInput;
```

Comment:
Might make it better to do

```suggestion
          payload: Exclude<AccountTransactionInput, UpdateContractInput | InitContractInput | SimpleTransferPayload>;
```